feature-assets/src/main/java/io/novafoundation/nova/feature_assets/presentation/balance/common/AssetGroupingDecoration.kt

```diff
@@ -111,13 +112,17 @@ class AssetGroupingDecoration(
         return getItemViewType(position)
     }
 
-    private fun isFinalItemInGroup(currentType: Int?, nextType: Int?): Boolean {
-        return currentType == GroupedListAdapter.TYPE_CHILD && (nextType == GroupedListAdapter.TYPE_GROUP || nextType == null)
+    private fun isFinalItemInGroup(nextType: Int?): Boolean {
+        return nextType == TYPE_NETWORK_GROUP || nextType == TYPE_TOKEN_GROUP || nextType == null
     }
 
     private fun shouldSkip(viewHolder: RecyclerView.ViewHolder): Boolean {
-        return viewHolder.bindingAdapterPosition == RecyclerView.NO_POSITION ||
-            (viewHolder !is AssetViewHolder && viewHolder !is AssetGroupViewHolder)
+        val isGroupViewHolder = viewHolder is NetworkAssetViewHolder ||
+            viewHolder is NetworkAssetGroupViewHolder ||
+            viewHolder is TokenAssetViewHolder ||
+            viewHolder is TokenAssetGroupViewHolder
```

Comment:
I suggest making a marker interface for all relevant ViewHolders like `AssetViewHolder` so that we dont need to modify this check in the future

Reply:
I fixed that in next prs